clap_derive/src/derives/attrs.rs

```diff
@@ -624,16 +624,13 @@ impl Attrs {
     pub fn has_explicit_methods(&self) -> bool {
         self.methods
             .iter()
-            .any(|m| m.name != "help" && m.name != "long_help")
+            .any(|m| m.name != "about" && m.name != "long_about")
     }
 
     pub fn has_doc_methods(&self) -> bool {
         !self.doc_comment.is_empty()
             || self.methods.iter().any(|m| {
-                m.name == "help"
-                    || m.name == "long_help"
-                    || m.name == "about"
-                    || m.name == "long_about"
+                m.name == "about" || m.name == "long_about"
```

Comment:
I added `help` and `long_help` back, but it still fails. Note that I had updated the test [help_is_better_than_comments](https://github.com/creativcoder/clap/blob/c79d9a561e550475e60ce029587b6cc710ac5b17/clap_derive/tests/doc-comments-help.rs#L43) case to use `about` attribute. So it shouldn't depend on `help`; right?

Reply:
You are right. I misunderstood the test. This code is correct.

Reply:
But at the same time: `top_long_doc_comment_both_help_long_help` test passes as it also uses `about` attribute, which is strange

Reply:
It uses a doc comment. Doc comments fly under `help` and `_long_help`. Replace this
https://github.com/clap-rs/clap/blob/3de8af46a2f0af709d91c3af23b24054cbf74f6a/clap_derive/src/derives/attrs.rs#L436

With `"about"` and it should work.

Reply:
I don't see anything wrong with `top_long_doc_comment_both_help_long_help`

Reply:
I will give this a look later. Could you please address the other issues for now?

Reply:
And you shouldn't! This comment 
```
        /// DO NOT PASS A BAR UNDER ANY CIRCUMSTANCES
        ///
        /// Or something else
```

translates to `long_help` (because it's multi paragraph) but attribute is `about` (without long).

Reply:
> It uses a doc comment. Doc comments fly under `help` and `long_help`. Replace this
> 
> https://github.com/clap-rs/clap/blob/3de8af46a2f0af709d91c3af23b24054cbf74f6a/clap_derive/src/derives/attrs.rs#L436
> 
> With `"about"` and it should work.

This has already been changed to "about" https://github.com/clap-rs/clap/blob/c79d9a561e550475e60ce029587b6cc710ac5b17/clap_derive/src/derives/attrs.rs#L436
It still fails.

Reply:
@CreepySkeleton ^

Reply:
I will take a look at this today and get back to you. Thanks for the patience.

Reply:
So, I found the issue. `Attrs::from_field` calls [this](https://github.com/clap-rs/clap/blob/master/clap_derive/src/derives/attrs.rs#L280) function. When `about` is encountered, instead of pushing a method, we are assigning it to `Attrs.about` [here](https://github.com/clap-rs/clap/blob/master/clap_derive/src/derives/attrs.rs#L344-L346). So, later when we are searching for `about` methods, we can't see the attribute because it's not a `Method` in `Attrs.methods`. You can fix this issue by removing `about` from `Attrs` and pushing the `about` attribute as a method when parsing it.

You would probably have to change the order of [these 2](https://github.com/clap-rs/clap/blob/master/clap_derive/src/derives/attrs.rs#L412-L413) lines afterwards. Similarly, with `Attrs::from_field`

Reply:
> You can fix this issue by removing about from Attrs and pushing the about attribute as a method when parsing it.

No wait! I was the one who wrote this code and thought I don't remember particulars, I remember it was about something important.

I'll see what is it about tomorrow.

Reply:
I checked a bit and as far as I can see, it wasn't really needed. @creativcoder should be coming up with any issues if they pop up.

@creativcoder Also, you need to edit the `benches` folder. They still have uses `Arg::help`.

Reply:
I remember now: it was about `no_version`. We removed this hack in clap_derive so it shuld be fine.